YakimS/assignment2/BinaryTree.java

```diff
@@ -0,0 +1,231 @@
+package CodeUAss1Package;
+
+import javax.management.openmbean.KeyAlreadyExistsException;
+import java.util.*;
+
+/**
+ * This class contains the root to a binary tree that can have only
+ * one node for each possible key.
+ * @param <T> is the type of data stored in the tree nodes value
+ */
+public class BinaryTree<T> {
+
+    private Node root;
+    private Set<Integer> keySet;
+
+    public BinaryTree() {
+        keySet = new HashSet<>();
+    }
+
+    public void printBinaryTree()
+    {
+        printBinaryTree(this.root,0);
+    }
+    private void printBinaryTree(Node root, int level){
+        if(root==null)

```

Comment:
Please avoid `if`s, `for`s etc without curly braces. It is hard to read and error-prone.